frontend/src/common/queries/collections.ts

```diff
@@ -502,3 +549,15 @@ export function useReuploadDataset(collectionId: string) {
     },
   });
 }
+
+/**
+ * Determine if a submitted DOI has failed validation on the BE. Expected response for invalid DOI:
+ * {"detail": "DOI cannot be found on Crossref", "status": 400, "title": "Bad Request", "type": "about:blank"}
+ * TODO generalize beyond DOI link type once all links are validated on the BE (#1916).
+ * @param status - Response status returned from server.
+ * @param detail - Response error text, if any.
+ * @returns True if DOI has been identified as invalid by the BE.
+ */
+function isInvalidDOI(status: number, detail?: string): boolean {
+  return status === 400 && detail === INVALID_DOI_MESSAGE;
```

Comment:
We have status enums!

https://github.com/chanzuckerberg/single-cell-data-portal/blob/b29ac1d746bd414ac44fe1959b52c97061eb4401/frontend/src/common/constants/HTTP_STATUS_CODE.ts

Reply:
Got it, thanks Timmy!